From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Thu, 24 Feb 2022 11:59:05 +0900
Subject: [PATCH] Purpur PR - Make Cauldron's fill chance for dripping lava and
 water configurable

Original by 12emin34
(PurpurMC Team)

Copyright (c) 2022 PurpurMC LLC

Permission is hereby granted, free of charge, to any person obtaining a
copy of this software and associated documentation files (the
"Software"), to deal in the Software without restriction, including
without limitation the rights to use, copy, modify, merge, publish,
distribute, sublicense, and/or sell copies of the Software, and to
permit persons to whom the Software is furnished to do so, subject to
the following conditions:

The above copyright notice and this permission notice shall be included
in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

diff --git a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
index 356b0cab78966f6b99396975c68042bff3526a9d..05165890179c0515a69e285342cbb28a06a3d47b 100644
--- a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
@@ -185,19 +185,19 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
 
     @VisibleForTesting
     public static void maybeFillCauldron(BlockState state, ServerLevel world, BlockPos pos, float dripChance) {
-        if (dripChance <= 0.17578125F || dripChance <= 0.05859375F) {
+        if (dripChance <= world.purpurConfig.cauldronDripstoneWaterFillChance || dripChance <= world.purpurConfig.cauldronDripstoneLavaFillChance) { // Prismarine - Purpur PR
             if (PointedDripstoneBlock.isStalactiteStartPos(state, world, pos)) {
                 Fluid fluidtype = PointedDripstoneBlock.getCauldronFillFluidType(world, pos);
                 float f1;
 
                 if (fluidtype == Fluids.WATER) {
-                    f1 = 0.17578125F;
+                    f1 = world.purpurConfig.cauldronDripstoneWaterFillChance; // Prismarine - Purpur PR
                 } else {
                     if (fluidtype != Fluids.LAVA) {
                         return;
                     }
 
-                    f1 = 0.05859375F;
+                    f1 = world.purpurConfig.cauldronDripstoneLavaFillChance; // Prismarine - Purpur PR
                 }
 
                 if (dripChance < f1) {
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index 3454d2ec2c76541a943c321477e2939cd20bab69..8e72cb27b4bee97658676de28071bfc84dd2dc29 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -3055,9 +3055,13 @@ public class PurpurWorldConfig {
 
     public float cauldronRainChance = 0.05F;
     public float cauldronPowderSnowChance = 0.1F;
+    public float cauldronDripstoneWaterFillChance = 0.17578125F; // Prismarine - Purpur PR
+    public float cauldronDripstoneLavaFillChance = 0.05859375F; // Prismarine - Purpur PR
     private void cauldronSettings() {
         cauldronRainChance = (float) getDouble("blocks.cauldron.fill-chances.rain", cauldronRainChance);
         cauldronPowderSnowChance = (float) getDouble("blocks.cauldron.fill-chances.powder-snow", cauldronPowderSnowChance);
+        cauldronDripstoneWaterFillChance = (float) getDouble("blocks.cauldron.fill-chances.dripstone-water", cauldronDripstoneWaterFillChance); // Prismarine - Purpur PR
+        cauldronDripstoneLavaFillChance = (float) getDouble("blocks.cauldron.fill-chances.dripstone-lava", cauldronDripstoneLavaFillChance); // Prismarine - Purpur PR
     }
 }
 
